src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/caseupdated/CaseUpdatedAboutToSubmitHandler.java

```diff
@@ -75,7 +80,54 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
 
         }
 
-        return new PreSubmitCallbackResponse<>(sscsCaseData);
+        PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(sscsCaseData);
+
+        validateAndUpdateDwpHandlingOffice(sscsCaseData,response);
+
+        return response;
+    }
+
+    private void validateAndUpdateDwpHandlingOffice(SscsCaseData sscsCaseData, PreSubmitCallbackResponse<SscsCaseData> response) {
+        DwpAddressLookupService dwpLookup = new DwpAddressLookupService();
+        MrnDetails mrnDetails = sscsCaseData.getAppeal().getMrnDetails();
+        BenefitType benefitType = sscsCaseData.getAppeal().getBenefitType();
+        boolean validBenefitType = validateBenefitType(benefitType,response);
+        boolean validDwpIssuingOffice = validateDwpIssuingOffice(mrnDetails, benefitType, response, dwpLookup);
+
+        if (validBenefitType && validDwpIssuingOffice) {
+            String regionalCenter = dwpLookup.getDwpRegionalCenterByBenefitTypeAndOffice(benefitType.getCode(), mrnDetails.getDwpIssuingOffice());
+            sscsCaseData.setDwpRegionalCentre(regionalCenter);
+        }
+    }
+
+    private boolean validateBenefitType(BenefitType benefitType, PreSubmitCallbackResponse<SscsCaseData> response) {
```

Comment:
I think as per requirement, if it updated by user then its need to validate and update dwpIssuingOffice
Jira ticket title when case is updated by user

Reply:
Caseupdate event is called by caseloader (auto) and also by user (manually)

Reply:
It is dwpRegionalCentre that need to be updated and only if both benefit type and dwpIssuingOffice is valid. That has been done, isn't it?